### PR TITLE
New post: Accelerating PyTorch Transformers with Intel Sapphire Rapids, part 2

### PR DESCRIPTION
**Pull request by juliensimon**
Follow up to https://huggingface.co/blog/intel-sapphire-rapids, covering inference this time

### CODE REVIEW

File: intel-sapphire-rapids-inference.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+For the sake of brevity, we'll just look at the p99 results for [distilbert-base-uncased](https://huggingface.co/distilbert-base-uncased). You'll find all other results at the end of the post. 
+
+<kbd>
+  <img src="assets/126_intel_sapphire_rapids_inference/01.png">
+</kbd>
+
+As you can see in the graph above, we get **35-40%** speedup out of the box, thanks to the combination of acceleration features present in the Sapphire Rapids architecture and in Hugging Face Optimum. 
+
+This lets us achieve reach **single-digit prediction latency** even with long text sequences, which was only possible with GPUs so far.
```
Comment (jeffboudier) [inline intel-sapphire-rapids-inference.md:168]:
```suggestion
This lets us achieve reach **single-digit millisecond prediction latency** even with long text sequences, which was only possible with GPUs so far.
```

**Review by jeffboudier — APPROVED** [review]
Great post! Btw this Space is a good playground to see the acceleration of SPR vs. ICX, although focused on stable diffusion so a bit off topic to this post and maybe not worth mentioning here: https://huggingface.co/spaces/Intel/Stable-Diffusion-Side-by-Side

**Comment by julien-c** [comment]
yes could be cool to mention it in passing!

File: intel-sapphire-rapids-inference.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+In this post, we're going to focus on inference. Working with popular HuggingFace transformers implemented with PyTorch, we'll first measure their performance on an Ice Lake server for short and long NLP token sequences. Then, we'll do the same with a Sapphire Rapids server and the latest version of Hugging Face [Optimum Intel](https://github.com/huggingface/optimum-intel), an open-source library dedicated to hardware acceleration for Intel platforms.
```
Comment (echarlaix) [inline intel-sapphire-rapids-inference.md:29]:
```suggestion
In this post, we're going to focus on inference. Working with popular Hugging Face [Transformers](https://github.com/huggingface/transformers) implemented with PyTorch, we'll first measure their performance on an Ice Lake server for short and long NLP token sequences. Then, we'll do the same with a Sapphire Rapids server and the latest version of Hugging Face [Optimum Intel](https://github.com/huggingface/optimum-intel), an open-source library dedicated to hardware acceleration for Intel platforms.
```

File: intel-sapphire-rapids-inference.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    result = benchmark(pipe, sentence_long_array)
+    print(f"Transformers pipeline, long sentence array: {result}")
+```
+
+On the r7iz (Sapphire Rapids) instance, we use both a vanilla pipeline and an Optimum pipeline in `bfloat16` mode to leverage the AMX instructions.
+
+
+```
+   import torch
+	from optimum.intel import inference_mode
```
Comment (echarlaix) [inline intel-sapphire-rapids-inference.md:147]:
```suggestion
   import torch
   from optimum.intel import inference_mode
```
Also I would  move these two imports outside the for loop (in the previous code snippet)

File: intel-sapphire-rapids-inference.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+	
+	with inference_mode(pipe, dtype=torch.bfloat16, verbose=False) as opt_pipe:
+	    result = benchmark(opt_pipe, sentence_short)
+	    print(f"Optimum pipeline, short sentence: {result}")
+	    result = benchmark(opt_pipe, sentence_long)
+	    print(f"Optimum pipeline, long sentence: {result}")
+	    result = benchmark(opt_pipe, sentence_short_array)
+	    print(f"Optimum pipeline, short sentence array: {result}")
+	    result = benchmark(opt_pipe, sentence_long_array)
+	    print(f"Optimum pipeline, long sentence array: {result}")
```
Comment (echarlaix) [inline intel-sapphire-rapids-inference.md:157]:
```suggestion
    with inference_mode(pipe, dtype=torch.bfloat16, verbose=False) as opt_pipe:
        result = benchmark(opt_pipe, sentence_short)
	print(f"Optimum pipeline, short sentence: {result}")
	result = benchmark(opt_pipe, sentence_long)
	print(f"Optimum pipeline, long sentence: {result}")
	result = benchmark(opt_pipe, sentence_short_array)
	print(f"Optimum pipeline, short sentence array: {result}")
	result = benchmark(opt_pipe, sentence_long_array)
	print(f"Optimum pipeline, long sentence array: {result}")
```

**Review by echarlaix — APPROVED** [review]
Great work @juliensimon !

**Comment by juliensimon** [comment]
Sent for review to Intel and AWS

**Comment by julien-c** [comment]
i updated this post to the new blog format, here: https://github.com/huggingface/blog/commit/12fcae031ccecd03f3fd4abf05622a6ada49160d

cc @osanseviero

**Pull request merged**